circleci: failed tests (13038): TestStatusServer

The following test appears to have failed:

[#13038](https://circleci.com/gh/cockroachdb/cockroach/13038):

```
I0219 06:24:39.896008 30154 acceptance/cluster/localcluster.go:437  *** started roach1.local ***
  ui:    https://0.0.0.0:32910
  trace: https://0.0.0.0:32910/debug/requests
  logs:  /tmp/circle-artifacts.z4yI3ZE/acceptance/TestStatusServer/roach1.local/cockroach.INFO
  pprof: docker exec -it 07c1f /bin/bash -c 'go tool pprof /cockroach <(wget --no-check-certificate -qO- https://$(hostname):26257/debug/pprof/heap)'
panic: test timed out after 5m0s

goroutine 2155 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:725 +0x14b
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0xe28b28, 0x12afbc0, 0xf, 0xf, 0xa7f69df4fb8d9701)
	/usr/local/go/src/testing/testing.go:583 +0x8d2
testing.(*M).Run(0xc823db8f08, 0xe27f60)
	/usr/local/go/src/testing/testing.go:515 +0x81
github.com/cockroachdb/cockroach/acceptance.TestMain(0xc823db8f08)
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:136 +0x3e
main.main()
	github.com/cockroachdb/cockroach/acceptance/_test/_testmain.go:80 +0x114

goroutine 17 [syscall, 4 minutes, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 5 [syscall, 4 minutes]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:116 +0x132
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x37

goroutine 6 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x12cc220)
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1032 +0x64
created by github.com/cockroachdb/cockroach/util/log.init.1
	/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:613 +0xbc

goroutine 9 [select, 4 minutes, locked to thread]:
runtime.gopark(0xe29340, 0xc820028f28, 0xcc07d8, 0x6, 0xc82011c118, 0x2)
	/usr/local/go/src/runtime/proc.go:262 +0x163
runtime.selectgoImpl(0xc820028f28, 0x0, 0x18)
	/usr/local/go/src/runtime/select.go:392 +0xa67
runtime.selectgo(0xc820028f28)
	/usr/local/go/src/runtime/select.go:215 +0x12
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal1_unix.go:279 +0x358
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1998 +0x1

goroutine 370 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba9886cc8, 0x72, 0xc820156000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820298680, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820298680, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820298620, 0xc820156000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122210, 0xc820156000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc820060680)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 21 [chan receive, 4 minutes]:
github.com/cockroachdb/cockroach/acceptance.TestMain.func1()
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:127 +0xd8
created by github.com/cockroachdb/cockroach/acceptance.TestMain
	/go/src/github.com/cockroachdb/cockroach/acceptance/main_test.go:135 +0x30

goroutine 2097 [select]:
github.com/docker/engine-api/client/transport/cancellable.Do(0x7f6ba9885ac0, 0xc820015db0, 0x7f6ba9885dc8, 0xc823cd1360, 0xc822550a80, 0x0, 0x0, 0x0)
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:56 +0x49d
github.com/docker/engine-api/client.(*Client).sendClientRequest(0xc821475860, 0x7f6ba9885ac0, 0xc820015db0, 0xcba1f0, 0x4, 0xc820d095c0, 0x52, 0x0, 0x7f6ba9885d38, 0xc8225b7c00, ...)
	/go/src/github.com/docker/engine-api/client/request.go:110 +0x45a
github.com/docker/engine-api/client.(*Client).sendRequest(0xc821475860, 0x7f6ba9885ac0, 0xc820015db0, 0xcba1f0, 0x4, 0xc820d095c0, 0x52, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/docker/engine-api/client/request.go:88 +0x2b8
github.com/docker/engine-api/client.(*Client).postWithContext(0xc821475860, 0x7f6ba9885ac0, 0xc820015db0, 0xc820d095c0, 0x52, 0x0, 0x0, 0x0, 0x0, 0x3, ...)
	/go/src/github.com/docker/engine-api/client/request.go:47 +0xb4
github.com/docker/engine-api/client.(*Client).post(0xc821475860, 0xc820d095c0, 0x52, 0x0, 0x0, 0x0, 0x0, 0xc820d095c0, 0x0, 0x0)
	/go/src/github.com/docker/engine-api/client/request.go:42 +0xce
--
testing.tRunner(0xc822bcef30, 0x12afce0)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892

goroutine 429 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8201664d0)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 25 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f6ba9886e48, 0x72, 0xc820213000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8201a8a00, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201a8a00, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8201a89a0, 0xc820213000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122770, 0xc820213000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc82024a270)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 26 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc82024a270)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 149 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f6ba9886b48, 0x72, 0xc820257000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820139020, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820139020, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820138fc0, 0xc820257000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a060, 0xc820257000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc8201aa9c0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1236 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc821b7a680)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 503 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba9886d88, 0x72, 0xc8203da000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc82024de90, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82024de90, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82024de30, 0xc8203da000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201222e8, 0xc8203da000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc82024a000)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 261 [IO wait, 4 minutes]:
net.runtime_pollWait(0x7f6ba9886c08, 0x72, 0xc82033a000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820138680, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820138680, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820138620, 0xc82033a000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201221f8, 0xc82033a000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc8200604e0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 400 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8201d3130)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 150 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc8201aa9c0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 504 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc82024a000)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 420 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8201d3340)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 412 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8202cb130)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 262 [select, 4 minutes]:
net/http.(*persistConn).writeLoop(0xc8200604e0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 371 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc820060680)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 415 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8202cbd90)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 449 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8202caa50)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 451 [chan receive, 4 minutes]:
database/sql.(*DB).connectionOpener(0xc8201d24d0)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 563 [chan receive, 3 minutes]:
database/sql.(*DB).connectionOpener(0xc8224e9a20)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 553 [chan receive, 3 minutes]:
database/sql.(*DB).connectionOpener(0xc8225b26e0)
	/usr/local/go/src/database/sql/sql.go:727 +0x45
created by database/sql.Open
	/usr/local/go/src/database/sql/sql.go:493 +0x33f

goroutine 633 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba9886608, 0x72, 0xc827954000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc825ec9720, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc825ec9720, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc825ec96c0, 0xc827954000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122648, 0xc827954000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc8200ed520)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 634 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc8200ed520)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1235 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f6ba98860c8, 0x72, 0xc821f56000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc821bcd6b0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc821bcd6b0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc821bcd650, 0xc821f56000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122e80, 0xc821f56000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc821b7a680)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1352 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc823b6c680)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1441 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f6ba9886008, 0x72, 0xc822aa4000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc820298ed0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820298ed0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc820298e70, 0xc822aa4000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122418, 0xc822aa4000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc822a7ab60)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 790 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc822a7bba0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1563 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f6ba9885f48, 0x72, 0xc822bd6000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8225a18e0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8225a18e0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8225a1880, 0xc822bd6000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a998, 0xc822bd6000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6c9c0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1128 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc821484410)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1026 [select, 3 minutes]:
net/http.(*persistConn).writeLoop(0xc823b6dba0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1009 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba9886308, 0x72, 0xc8239f0000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc823dc8d80, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc823dc8d80, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc823dc8d20, 0xc8239f0000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a990, 0xc8239f0000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6dba0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1458 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc822a7ab60)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 789 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba9886188, 0x72, 0xc82355c000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc822c513a0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc822c513a0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc822c51340, 0xc82355c000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc8201228f8, 0xc82355c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc822a7bba0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1127 [IO wait, 3 minutes]:
net.runtime_pollWait(0x7f6ba98863c8, 0x72, 0xc82147b000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8215f46f0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8215f46f0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8215f4690, 0xc82147b000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002aac8, 0xc82147b000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc821484410)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1564 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc823b6c9c0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1351 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f6ba9886248, 0x72, 0xc8217c5000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8225a1800, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8225a1800, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8225a17a0, 0xc8217c5000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a2d8, 0xc8217c5000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6c680)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1668 [select, 1 minutes]:
net/http.(*persistConn).writeLoop(0xc823b6d040)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 2116 [IO wait]:
net.runtime_pollWait(0x7f6ba404d0e0, 0x72, 0xc822b4c000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8205e1790, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8205e1790, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8205e1730, 0xc822b4c000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820123848, 0xc822b4c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6cdd0)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1667 [IO wait, 1 minutes]:
net.runtime_pollWait(0x7f6ba404d3e0, 0x72, 0xc823d04000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc823adcbc0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc823adcbc0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc823adcb60, 0xc823d04000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a360, 0xc823d04000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6d040)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 2127 [select]:
net/http.(*persistConn).readLoop(0xc822a7b450)
	/usr/local/go/src/net/http/transport.go:1162 +0xd52
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 2128 [select]:
net/http.(*persistConn).writeLoop(0xc822a7b450)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 2130 [select]:
github.com/docker/engine-api/client/transport/cancellable.Do.func3(0x7f6ba98c5870, 0xc8224fa700, 0xc822659490, 0xc820b38000)
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:78 +0xef
created by github.com/docker/engine-api/client/transport/cancellable.Do
	/go/src/github.com/docker/engine-api/client/transport/cancellable/cancellable.go:84 +0x336

goroutine 1769 [IO wait]:
net.runtime_pollWait(0x7f6ba404d320, 0x72, 0xc823d1c000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8205e1250, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8205e1250, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8205e11f0, 0xc823d1c000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a3d0, 0xc823d1c000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6cc30)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1770 [select]:
net/http.(*persistConn).writeLoop(0xc823b6cc30)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 2117 [select]:
net/http.(*persistConn).writeLoop(0xc823b6cdd0)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1986 [IO wait]:
net.runtime_pollWait(0x7f6ba404d1a0, 0x72, 0xc8225b8000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc82010c300, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82010c300, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82010c2a0, 0xc8225b8000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a058, 0xc8225b8000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc823b6c270)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 1879 [IO wait]:
net.runtime_pollWait(0x7f6ba404d260, 0x72, 0xc8225ec000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc82255a680, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc82255a680, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc82255a620, 0xc8225ec000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc820122260, 0xc8225ec000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
net/http.(*persistConn).readLoop(0xc822a7a340)
	/usr/local/go/src/net/http/transport.go:1053 +0x177
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:837 +0x10a6

goroutine 2123 [IO wait]:
net.runtime_pollWait(0x7f6ba404d020, 0x72, 0xc820254000)
	/usr/local/go/src/runtime/netpoll.go:160 +0x60
net.(*pollDesc).Wait(0xc8225b6ed0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8225b6ed0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).Read(0xc8225b6e70, 0xc820254000, 0x1000, 0x1000, 0x0, 0x7f6ba98bd050, 0xc8200141e0)
	/usr/local/go/src/net/fd_unix.go:250 +0x23a
net.(*conn).Read(0xc82002a170, 0xc820254000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/net.go:172 +0xe4
--
github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).monitor(0xc822a9d6b0)
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:491 +0x215
created by github.com/cockroachdb/cockroach/acceptance/cluster.(*LocalCluster).Start
	/go/src/github.com/cockroachdb/cockroach/acceptance/cluster/localcluster.go:525 +0x3aa

goroutine 1880 [select]:
net/http.(*persistConn).writeLoop(0xc822a7a340)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 1987 [select]:
net/http.(*persistConn).writeLoop(0xc823b6c270)
	/usr/local/go/src/net/http/transport.go:1257 +0x472
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:838 +0x10cb

goroutine 2154 [select]:
net/http.(*persistConn).roundTrip(0xc823b6cdd0, 0xc825ebe660, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1453 +0xf1f
net/http.(*Transport).RoundTrip(0xc8205da600, 0xc822550a80, 0xc8205da600, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:304 +0x9bb
net/http.send(0xc822550a80, 0x7f6ba98c1858, 0xc8205da600, 0x0, 0x0, 0x0, 0xc8225ba900, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:260 +0x6b7
net/http.(*Client).send(0xc8239e2900, 0xc822550a80, 0x0, 0x0, 0x0, 0xc820d08f58, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:155 +0x185
net/http.(*Client).doFollowingRedirects(0xc8239e2900, 0xc822550a80, 0xe28ec0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:475 +0x8a4

```
Please assign, take a look and update the issue accordingly.

--- a/acceptance/cluster/docker.go
+++ b/acceptance/cluster/docker.go
@@ -28,9 +28,12 @@ import (
 	"runtime"
 	"strconv"
 	"strings"
+	"time"
 
+	"github.com/docker/engine-api/client"
 	"github.com/docker/engine-api/types"
 	"github.com/docker/engine-api/types/container"
+	"github.com/docker/engine-api/types/network"
 	"github.com/docker/go-connections/nat"
 	"golang.org/x/net/context"
 
@@ -134,19 +137,11 @@ func maybePanic(err error) {
 // Remove removes the container from docker. It is an error to remove a running
 // container.
 func (c *Container) Remove() error {
-	err := c.cluster.client.ContainerRemove(context.Background(), types.ContainerRemoveOptions{
+	return c.cluster.client.ContainerRemove(context.Background(), types.ContainerRemoveOptions{
 		ContainerID:   c.id,
 		RemoveVolumes: true,
 		Force:         true,
 	})
-
-	if os.Getenv("CIRCLECI") == "true" {
-		// HACK: Removal of docker containers on circleci reports the error:
-		// "Driver btrfs failed to remove root filesystem". So if we're running on
-		// circleci, just ignore the error, the containers are still removed.
-		return nil
-	}
-	return err
 }
 
 // Kill stops a running container, without removing it.
@@ -282,4 +277,171 @@ func (c *Container) Addr(port nat.Port) *net.TCPAddr {
 		IP:   dockerIP(),
 		Port: portNum,
 	}
+}
+
+// resilientDockerClient handles certain recoverable Docker usage errors.
+//
+// For example, `ContainerCreate` will fail if a container with the requested
+// name already exists. resilientDockerClient will catch this, delete the
+// existing container and try again.
+type resilientDockerClient struct {
+	client.APIClient
+}
+
+func (cli resilientDockerClient) ContainerCreate(
+	ctx context.Context, config *container.Config, hostConfig *container.HostConfig,
+	networkingConfig *network.NetworkingConfig, containerName string,
+) (types.ContainerCreateResponse, error) {
+	response, err := cli.APIClient.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
+	if err != nil && strings.Contains(err.Error(), "already in use") {
+		log.Infof("unable to create container %s: %v", containerName, err)
+		containers, cerr := cli.ContainerList(ctx, types.ContainerListOptions{All: true})
+		if cerr != nil {
+			log.Infof("unable to list containers: %v", cerr)
+			return types.ContainerCreateResponse{}, err
+		}
+		for _, c := range containers {
+			for _, n := range c.Names {
+				// The container names begin with a "/".
+				n = strings.TrimPrefix(n, "/")
+				if n != containerName {
+					continue
+				}
+				log.Infof("Trying to remove %s", c.ID)
+				roptions := types.ContainerRemoveOptions{
+					ContainerID:   c.ID,
+					RemoveVolumes: true,
+					Force:         true,
+				}
+				if rerr := cli.ContainerRemove(ctx, roptions); rerr != nil {
+					log.Infof("unable to remove container: %v", rerr)
+					return types.ContainerCreateResponse{}, err
+				}
+				return cli.ContainerCreate(ctx, config, hostConfig, networkingConfig, containerName)
+			}
+		}
+		return types.ContainerCreateResponse{}, err
+	}
+	return response, err
+}
+
+func (cli resilientDockerClient) ContainerRemove(ctx context.Context, options types.ContainerRemoveOptions) error {
+	err := cli.APIClient.ContainerRemove(ctx, options)
+	// TODO(dan): Is the "No such container" error happening in practice?
+	if err != nil && strings.Contains(err.Error(), "No such container") {
+		return nil
+	}
+
+	if os.Getenv("CIRCLECI") == "true" {
+		// HACK: Removal of docker containers on circleci reports the error:
+		// "Driver btrfs failed to remove root filesystem". So if we're running on
+		// circleci, just ignore the error, the containers are still removed.
+		return nil
+	}
+	return err
+}
+
+// retryingDockerClient proxies the Docker client api and retries problematic
+// calls.
+//
+// Sometimes http requests to the Docker server, on circleci in particular, will
+// hang indefinitely and non-deterministically. This leads to flaky tests. To
+// avoid this, we wrap some of them in a timeout and retry loop.
+type retryingDockerClient struct {
+	client.APIClient
+	attempts int
+	timeout  time.Duration
+}
+
+func (cli retryingDockerClient) ContainerCreate(
+	ctx context.Context, config *container.Config, hostConfig *container.HostConfig,
+	networkingConfig *network.NetworkingConfig, containerName string,
+) (types.ContainerCreateResponse, error) {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		ret, err := cli.APIClient.ContainerCreate(timeoutCtx, config, hostConfig, networkingConfig, containerName)
+		if err == context.DeadlineExceeded {
+			continue
+		}
+		return ret, err
+	}
+	err := fmt.Errorf("exceeded %d tries of ContainerCreate with a %s timeout", cli.attempts, cli.timeout)
+	return types.ContainerCreateResponse{}, err
+}
+
+func (cli retryingDockerClient) ContainerStart(ctx context.Context, containerID string) error {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		err := cli.APIClient.ContainerStart(timeoutCtx, containerID)
+		if err == nil {
+			return nil
+		} else if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
+			continue
+		} else {
+			return err
+		}
+	}
+	return fmt.Errorf("exceeded %d tries of ContainerStart with a %s timeout", cli.attempts, cli.timeout)
+}
+
+func (cli retryingDockerClient) ContainerRemove(ctx context.Context, options types.ContainerRemoveOptions) error {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		err := cli.APIClient.ContainerRemove(timeoutCtx, options)
+		if err == nil {
+			return nil
+		} else if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
+			continue
+		} else {
+			return err
+		}
+	}
+	return fmt.Errorf("exceeded %d tries of ContainerRemove with a %s timeout", cli.attempts, cli.timeout)
+}
+
+func (cli retryingDockerClient) ContainerKill(ctx context.Context, containerID, signal string) error {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		err := cli.APIClient.ContainerKill(timeoutCtx, containerID, signal)
+		if err == nil {
+			return nil
+		} else if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
+			continue
+		} else {
+			return err
+		}
+	}
+	return fmt.Errorf("exceeded %d tries of ContainerKill with a %s timeout", cli.attempts, cli.timeout)
+}
+
+func (cli retryingDockerClient) ImagePull(
+	ctx context.Context, options types.ImagePullOptions, privilegeFunc client.RequestPrivilegeFunc,
+) (io.ReadCloser, error) {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		ret, err := cli.APIClient.ImagePull(timeoutCtx, options, privilegeFunc)
+		if err == nil {
+			return ret, nil
+		} else if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
+			continue
+		} else {
+			return nil, err
+		}
+	}
+	return nil, fmt.Errorf("exceeded %d tries of ImagePull with a %s timeout", cli.attempts, cli.timeout)
+}
+
+func (cli retryingDockerClient) ContainerWait(ctx context.Context, containerID string) (int, error) {
+	for i := 0; i < cli.attempts; i++ {
+		timeoutCtx, _ := context.WithTimeout(ctx, cli.timeout)
+		ret, err := cli.APIClient.ContainerWait(timeoutCtx, containerID)
+		if err == nil {
+			return ret, nil
+		} else if strings.Contains(err.Error(), context.DeadlineExceeded.Error()) {
+			continue
+		} else {
+			return 0, err
+		}
+	}
+	return 0, fmt.Errorf("exceeded %d tries of ContainerWait with a %s timeout", cli.attempts, cli.timeout)
 }

--- a/acceptance/cluster/localcluster.go
+++ b/acceptance/cluster/localcluster.go
@@ -125,7 +125,7 @@ type testNode struct {
 // cluster is composed of a "volumes" container which manages the
 // persistent volumes used for certs and node data and N cockroach nodes.
 type LocalCluster struct {
-	client               *client.Client
+	client               client.APIClient
 	stopper              chan struct{}
 	mu                   sync.Mutex // Protects the fields below
 	vols                 *Container
@@ -159,8 +159,15 @@ func CreateLocal(cfg TestConfig, logDir string, stopper chan struct{}) *LocalClu
 	cli, err := client.NewEnvClient()
 	maybePanic(err)
 
+	resilientClient := resilientDockerClient{APIClient: cli}
+	retryingClient := retryingDockerClient{
+		APIClient: resilientClient,
+		attempts:  3,
+		timeout:   time.Minute,
+	}
+
 	return &LocalCluster{
-		client:  cli,
+		client:  retryingClient,
 		stopper: stopper,
 		config:  cfg,
 		// TODO(tschottdorf): deadlocks will occur if these channels fill up.
@@ -354,9 +361,12 @@ func (l *LocalCluster) initCluster() {
 		"volumes",
 	)
 	maybePanic(err)
+	// Make sure this assignment to l.vols is before the calls to Start and Wait.
+	// Otherwise, if they trigger maybePanic, this container won't get cleaned up
+	// and it'll get in the way of future runs.
+	l.vols = c
 	maybePanic(c.Start())
 	maybePanic(c.Wait())
-	l.vols = c
 }
 
 func (l *LocalCluster) createRoach(node *testNode, vols *Container, env []string, cmd ...string) {

